frontend/src/core/static/files.ts

```diff
@@ -79,6 +79,16 @@ export function patchVegaLoader(
 
   function maybeGetVirtualFile(url: string): string | undefined {
     const pathname = new URL(url, document.baseURI).pathname;
+
+    // If if the URL starts with file://, then using the document.baseURI
+    // will not work. In this case, should just chop off from /@file/...
+    if (url.startsWith("file://")) {
```

Comment:
I couldn't see how this change is related? Looks fine, was just wondering

Reply:
its not related but i noticed it when trying to test out this change. basically the patch doesn't work from opening a local html file. (only when served from http(s))